templates/deployment.yaml

```diff
@@ -53,10 +53,8 @@ spec:
                   key: {{ .existingSecret.passwordKey }}
             {{- end }}
             {{- end }}
-            {{- with .Values.shadowsocks.dnsServers }}
             - name: DNS_ADDRS
```

Comment:
I think we should probably surround this part with `{{- if .Values.shadowsocks.dnsServers }} ... {{- end }}`. Passing an empty `DNS_ADDRS` value to the `shadowsocks/shadowsocks-libev` image would get the following error:

```
2021-12-16 14:08:40 ERROR: failed to set nameservers
```

Reply:
@predatorray You are right, there is a bug in the `shadowsocks/shadowsocks-libev` docker image. For the moment you are not able to use an empty `DNS_ADDRS` value, which means you are not able to use Kubernetes's internal DNS server (the one that is already setup in `/etc/resolv.conf`). I've submitted a fix on the same day that I've opened this PR, but it hasn't been merged yet: https://github.com/shadowsocks/shadowsocks-libev/pull/2866. I will keep you posted!

Reply:
The solution you suggested won't work. Let me rephrase: the goal of this PR is to be able to set an empty string for `DNS_ADDRS` in order to use Kubernetes' internal DNS server, instead of the default `8.8.8.8`. Doing so will allow shadowsocks proxy users to access any of the cluster's services, like a VPN hosted on the cluster for instance.

To use Kubernetes' internal DNS server, I need to set the `DNS_ADDRS` env var as an empty string. That way, the `-d` option of `ss-server` won't be used, and the DNS server IP address will be fetched from _/etc/resolv.conf_.  See [the documentation of ss-server](https://github.com/shadowsocks/shadowsocks-libev/blob/89b5f987d6a5329de9713704615581d363f0cfed/doc/ss-server.asciidoc):

> **-d \<addr\>**
>        Setup name servers for internal DNS resolver (libc-ares). **The default server is fetched from _/etc/resolv.conf_.**

The bug is that, as of today, even if you pass an empty string, the `-d` option is still used, so you get
```
ERROR: failed to set nameservers.
```
This will be fixed in https://github.com/shadowsocks/shadowsocks-libev/pull/2866

Reply:
Thanks for your explanation! Yes, this is indeed a problem. The image does not provide any mechanism to reset the `-d` option to its default behavior but instead using `8.8.8.8,8.8.4.4` as its default value. Let's see what is the suggestion from the maintainer.

Reply:
Hi @predatorray! The fix for the Docker image was merged in https://github.com/shadowsocks/shadowsocks-libev/pull/2866. The proposed change for the Helm Chart is now relevant 🙂 I am using it in production, and it works like a charm!

Reply:
Thanks @alexandre-abrioux for your efforts to push this feature forward. Considering it as an incompatible change, since setting `DNS_ADDRS` to be an empty will fail using the latest stable image, let's wait for the release of a new stable image for this and change the `appVersion` in the `Chart.yaml` accordingly.